mantidproject/mantid
# Bugs in DrILL interface

**Description of work.**

This PR solves several bugs in DrILL interface:
1. ILL data directory on Linux
2. remove of leading an trailing spaces in `CustomOptions`
3. automatic export when the default save directory setting is not set
4. automatic increment with special value 0

**To test:**

1. code review
2. try to write something like `Clear2DWorkspace=False ; ThetaDependent=False ` in the CustomOptions column in DrILL
3. clear the default save directory setting, process some data through DrILL and observe the logs
4. increment/decrement with 0 in the cell. Decrement from a number to see that 0 is never reached

Part of #30592 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
